pkg/kubeapiserver/options/authentication.go

```diff
@@ -639,11 +643,64 @@ func (o *BuiltInAuthenticationOptions) ApplyTo(authInfo *genericapiserver.Authen
 	}
 
 	// var openAPIV3SecuritySchemes spec3.SecuritySchemes
-	authenticator, openAPIV2SecurityDefinitions, openAPIV3SecuritySchemes, err := authenticatorConfig.New()
+	authenticator, updateAuthenticationConfig, openAPIV2SecurityDefinitions, openAPIV3SecuritySchemes, err := authenticatorConfig.New(ctx)
 	if err != nil {
 		return err
 	}
 	authInfo.Authenticator = authenticator
+
+	if len(o.AuthenticationConfigFile) > 0 {
+		trackedAuthenticationConfigHash := authenticatorConfig.AuthenticationConfigHash
+		var mu sync.Mutex
+		go filesystem.WatchUntil(
+			ctx,
+			time.Minute,
+			o.AuthenticationConfigFile,
+			func() {
+				// TODO add metrics
+				// TODO collapse onto shared logic with DynamicEncryptionConfigContent controller
+
+				mu.Lock()
+				defer mu.Unlock()
+
+				// scope currentHash to this block because we only want to track the hash from the load func
+				{
+					currentHash, err := getAuthenticationConfigHash(o.AuthenticationConfigFile)
+					if err != nil {
+						klog.ErrorS(err, "failed to get authentication config")
+						return
+					}
+
+					if currentHash == trackedAuthenticationConfigHash {
+						return
+					}
+				}
+
+				authConfig, authConfigHash, err := loadAuthenticationConfig(o.AuthenticationConfigFile)
```

Comment:
both loadAuthenticationConfig and getAuthenticationConfigHash call readAuthenticationConfigDataAndHash on the same file. currentHash and authConfigHash are the same in case currentHash != trackedAuthenticationConfigHash.

i guess it's not perf critical, but the hash can be computed once if instead of loadAuthenticationConfig there are 2 actions - read, parse. but no hash compute, as was it was done earlier.